apps/els_lsp/src/els_code_lens_suggest_spec.erl

```diff
@@ -30,6 +30,9 @@ init(#{uri := Uri} = _Document) ->
         Info ->
             Info
     catch
+        throw:{dialyzer_error, "Could not read PLT file " ++ _ = Message} ->
```

Comment:
I'm not sure about this, since we would be losing the original message and stacktrace. If we want to make it clear to users, we could point them to the config item they are supposed to change or provide an example. What do you think?

Reply:
Before trying to identify the errors, I made some tests using a main branch build. Log messages from `els_dialyzer_diagnostics` are included for comparison, since it also depends on dialyzer, like the spec lens does.

These logs, together with the little hint in the editor "Cannot extract specs (check logs for details)", and the logs saying "Cannot extract typer info." together with a lot of other text was the reason I thought the logging could be improved. Finding the details was not so simple.

With my latest changes I replaced the lens state `no_info` with an `{error, Reason}` tuple, to get a short tag to show beside "check logs for details" so it's easier to tie the issues together.

### With old PLT file

I don't know why `els_dialyzer_diagnostics` log messages show up multiple times, out of scope for my changes in this PR.

```
[2022-11-08T21:11:50.430878+01:00] [error] Error while running dialyzer [type=throw] [error={dialyzer_error,"File not found: /usr/local/Cellar/erlang/25.0_1/lib/erlang/lib/crypto-5.1/ebin/crypto.beam\n"}] [els_dialyzer_diagnostics:run/1 L61] <0.173.0>
[Error - 9:11:50 PM] [2022-11-08T21:11:50.430878+01:00] [error] Error while running dialyzer [type=throw] [error={dialyzer_error,"File not found: /usr/local/Cellar/erlang/25.0_1/lib/erlang/lib/crypto-5.1/ebin/crypto.beam\n"}] [els_dialyzer_diagnostics:run/1 L61] <0.173.0>

[Warn  - 9:11:51 PM] [2022-11-08T21:11:51.159076+01:00] [warning] Cannot extract typer info., Class: throw, Exception: {dialyzer_error,"Old PLT file /Users/fabian/.cache/rebar3/rebar3_25.0_plt\n"}, Stacktrace: [{dialyzer_plt,plt_error,1,[{file,"dialyzer_plt.erl"},{line,669}]},{els_typer,collect_info,1,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_typer.erl"},{line,303}]},{els_typer,get_info,1,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_typer.erl"},{line,84}]},{els_code_lens_suggest_spec,init,1,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_code_lens_suggest_spec.erl"},{line,29}]},{els_code_lens,lenses,2,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_code_lens.erl"},{line,91}]},{els_code_lens_provider,'-run_lenses_job/1-lc$^0/1-0-',3,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_code_lens_provider.erl"},{line,37}]},{els_code_lens_provider,'-run_lenses_job/1-lc$^0/1-0-',3,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_code_lens_provider.erl"},{line,38}]},{els_code_lens_provider,'-run_lenses_job/1-fun-1-',2,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_code_lens_provider.erl"},{line,38}]}] [els_code_lens_suggest_spec:init/1 L39] <0.406.0>
```

### With bad `plt_path`

```
[2022-11-08T21:13:12.044639+01:00] [error] Error while running dialyzer [type=throw] [error={dialyzer_error,"No such file, directory or application: \"/Users/fabian/.cache/rebar3/rebar3_s25.0_plt\""}] [els_dialyzer_diagnostics:run/1 L61] <0.173.0>

[Error - 9:13:12 PM] [2022-11-08T21:13:12.044639+01:00] [error] Error while running dialyzer [type=throw] [error={dialyzer_error,"No such file, directory or application: \"/Users/fabian/.cache/rebar3/rebar3_s25.0_plt\""}] [els_dialyzer_diagnostics:run/1 L61] <0.173.0>

[Warn  - 9:13:12 PM] [2022-11-08T21:13:12.894359+01:00] [warning] Cannot extract typer info., Class: throw, Exception: {dialyzer_error,"Could not read PLT file /Users/fabian/.cache/rebar3/rebar3_s25.0_plt: no_such_file\n"}, Stacktrace: [{dialyzer_plt,plt_error,1,[{file,"dialyzer_plt.erl"},{line,669}]},{els_typer,collect_info,1,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_typer.erl"},{line,303}]},{els_typer,get_info,1,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_typer.erl"},{line,84}]},{els_code_lens_suggest_spec,init,1,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_code_lens_suggest_spec.erl"},{line,29}]},{els_code_lens,lenses,2,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_code_lens.erl"},{line,91}]},{els_code_lens_provider,'-run_lenses_job/1-lc$^0/1-0-',3,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_code_lens_provider.erl"},{line,37}]},{els_code_lens_provider,'-run_lenses_job/1-lc$^0/1-0-',3,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_code_lens_provider.erl"},{line,38}]},{els_code_lens_provider,'-run_lenses_job/1-fun-1-',2,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_code_lens_provider.erl"},{line,38}]}] [els_code_lens_suggest_spec:init/1 L39] <0.406.0>
```

### With default `plt_path`

```
[Warn  - 9:15:56 PM] [2022-11-08T21:15:56.229187+01:00] [warning] Cannot extract typer info., Class: throw, Exception: {dialyzer_error,"Could not read PLT file /Users/fabian/Library/Caches/erlang/.dialyzer_plt: no_such_file\n"}, Stacktrace: [{dialyzer_plt,plt_error,1,[{file,"dialyzer_plt.erl"},{line,669}]},{els_typer,collect_info,1,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_typer.erl"},{line,303}]},{els_typer,get_info,1,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_typer.erl"},{line,84}]},{els_code_lens_suggest_spec,init,1,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_code_lens_suggest_spec.erl"},{line,29}]},{els_code_lens,lenses,2,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_code_lens.erl"},{line,91}]},{els_code_lens_provider,'-run_lenses_job/1-lc$^0/1-0-',3,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_code_lens_provider.erl"},{line,37}]},{els_code_lens_provider,'-run_lenses_job/1-lc$^0/1-0-',3,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_code_lens_provider.erl"},{line,38}]},{els_code_lens_provider,'-run_lenses_job/1-fun-1-',2,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_code_lens_provider.erl"},{line,38}]}] [els_code_lens_suggest_spec:init/1 L39] <0.372.0>
```

### With not a PLT file at `plt_path`

```
[Error - 9:25:45 PM] [2022-11-08T21:25:45.784494+01:00] [error] Error while running dialyzer [type=throw] [error={dialyzer_error,"The file: /Users/fabian/x.txt is not a valid PLT file\nUse the options:\n   --build_plt   to build a new PLT; or\n   --add_to_plt  to add to an existing PLT\n\nFor example, use a command like the following:\n   dialyzer --build_plt --apps erts kernel stdlib mnesia\nNote that building a PLT such as the above may take 20 mins or so\n\nIf you later need information about other applications, say crypto,\nyou can extend the PLT by the command:\n  dialyzer --add_to_plt --apps crypto\nFor applications that are not in Erlang/OTP use an absolute file name.\n"}] [els_dialyzer_diagnostics:run/1 L61] <0.173.0>

[2022-11-08T21:25:45.784494+01:00] [error] Error while running dialyzer [type=throw] [error={dialyzer_error,"The file: /Users/fabian/x.txt is not a valid PLT file\nUse the options:\n   --build_plt   to build a new PLT; or\n   --add_to_plt  to add to an existing PLT\n\nFor example, use a command like the following:\n   dialyzer --build_plt --apps erts kernel stdlib mnesia\nNote that building a PLT such as the above may take 20 mins or so\n\nIf you later need information about other applications, say crypto,\nyou can extend the PLT by the command:\n  dialyzer --add_to_plt --apps crypto\nFor applications that are not in Erlang/OTP use an absolute file name.\n"}] [els_dialyzer_diagnostics:run/1 L61] <0.173.0>

[Warn  - 9:25:46 PM] [2022-11-08T21:25:46.108560+01:00] [warning] Cannot extract typer info., Class: throw, Exception: {dialyzer_error,"Could not read PLT file /Users/fabian/x.txt: not_valid\n"}, Stacktrace: [{dialyzer_plt,plt_error,1,[{file,"dialyzer_plt.erl"},{line,669}]},{els_typer,collect_info,1,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_typer.erl"},{line,303}]},{els_typer,get_info,1,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_typer.erl"},{line,84}]},{els_code_lens_suggest_spec,init,1,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_code_lens_suggest_spec.erl"},{line,29}]},{els_code_lens,lenses,2,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_code_lens.erl"},{line,91}]},{els_code_lens_provider,'-run_lenses_job/1-lc$^0/1-0-',3,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_code_lens_provider.erl"},{line,37}]},{els_code_lens_provider,'-run_lenses_job/1-lc$^0/1-0-',3,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_code_lens_provider.erl"},{line,38}]},{els_code_lens_provider,'-run_lenses_job/1-fun-1-',2,[{file,"/Users/fabian/code/fabjan/erlang_ls/apps/els_lsp/src/els_code_lens_provider.erl"},{line,38}]}] [els_code_lens_suggest_spec:init/1 L39] <0.401.0>
```